src/utils/formats.ts

```diff
@@ -1,4 +1,4 @@
-const HAIKU_REGEX = /^[a-z]+-[a-z]+-\d+$/;
+const HAIKU_REGEX = /^[a-z]+-[a-z]+-\d+(-\w+)?$/;
```

Comment:
I changed the format based on feedback. Now the third part of ID is alphanumeric, the length is the same. Sorry for the back and forth, this is the merged version.
```suggestion
const HAIKU_REGEX = /^[a-z]+-[a-z]+-[a-z0-9]+$/;
```